envbuilder.go

```diff
@@ -342,6 +343,21 @@ func Run(ctx context.Context, options Options) error {
 		}
 	}
 
+	if options.PushImage {
+		// Copy the envbuilder binary into the build context.
+		buildParams.DockerfileContent = buildParams.DockerfileContent + "\n" +
+			fmt.Sprintf("COPY %s %s", ".envbuilder", "/.envbuilder")
+
+		log.Println("FIXME show me the Dockerfile now: " + buildParams.DockerfileContent)
+
+		binPath := filepath.Join(MagicDir, "bin", "envbuilder")
+		dst := filepath.Join(buildParams.BuildContext, binPath)
+		err := copyFile(binPath, dst)
```

Comment:
@BrunoQuaresma @johnstcn 

Alright, I spent some time digging into this. I don't get one thing about the Docker lifecycle: 

```
    integration_test.go:1633: "#2: COPY .envbuilder /.envbuilder"
    integration_test.go:1633: "Failed to build: do build: error building stage: failed to execute command: copying dir: creating file: open /.envbuilder/bin/envbuilder: text file busy"
    integration_test.go:1633: "Falling back to the default image..."
    integration_test.go:1633: "error: do build: error building stage: failed to execute command: copying dir: creating file: open /.envbuilder/bin/envbuilder: text file busy: no fallback image has been specified"
    integration_test.go:1383:
        	Error Trace:	/code/envbuilder/integration/integration_test.go:1383
        	Error:      	Received unexpected error:
        	            	error: do build: error building stage: failed to execute command: copying dir: creating file: open /.envbuilder/bin/envbuilder: text file busy: no fallback image has been specified
        	Test:       	TestEmbedBinaryImage
--- FAIL: TestEmbedBinaryImage (0.62s)
```

As you see `COPY` failed because the binary, which is already there, has been started? And the binary is there because in the integration test we use `localhost:5000/envbuilder-test-alpine:latest`. I see we have 2 options:
1. Do not use Docker image with `envbuilder` as base.
2. Check if binary is there and do not copy it.

Side question: why is the binary started? Is it always like that with kaniko?

Reply:
> Do not use Docker image with envbuilder as base.

The image `localhost:5000/envbuilder-test-alpine:latest` should not contain the `envbuilder` binary. It simply a locally-cached version of `alpine:latest`.

```
docker run -it --rm localhost:5000/envbuilder-test-alpine:latest ls -la /.envbuilder
ls: /.envbuilder: No such file or directory
```

> Side question: why is the binary started?

The `envbuilder` binary is going to be running inside the container no matter what, as that's the container entrypoint.

Reply:
OK, so it seems that `COPY ` operation is not necessary? See the build fails due to `text file busy`. This is the part I'm trying to understand.

Reply:
OK, instead of copying the binary manually, we need to instead tell Kaniko to not skip over `/.envbuilder/bin/envbuilder`.

Reply:
@johnstcn just double checking, are you going to take over this PR?

Reply:
yep